### PR TITLE
[CHK-1178] refactor(closepayment): send refund request event instead of closepayment event when getting error or KO from closePayment

### PR DESCRIPTION
**Pull request by giovanniberti**
#### List of Changes

* send refund request event instead of closure after bad or KO `closePayment`
* save CLOSURE_ERROR instead of CLOSURE event when getting KO or unrecoverable Nodo error on `closePayment`

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

Depends on #160

### CODE REVIEW

File: src/main/java/it/pagopa/transactions/commands/handlers/TransactionSendClosureHandler.java [hunk 1]
```diff
@@ -297,26 +297,53 @@ private ClosePaymentRequestV2Dto.OutcomeEnum authorizationResultToOutcomeV2(
...
-                .filter(e ->
-                // Closed event sent on the queue only if the transaction was previously
-                // authorized and the Nodo response outcome is KO
-                TransactionClosureData.Outcome.KO.equals(e.getData().getResponseOutcome())
-                        && AuthorizationResultDto.OK.equals(authorizationResult)
+        return Mono.just(closureOutcomeEvent)
+                .filter(
+                        e -> e.fold(
+                                closureErrorEvent -> true,
+                                // Closed event sent on the queue only if the transaction was previously
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/commands/handlers/TransactionSendClosureHandler.java:308]:
```suggestion
                                // Refund requested event sent on the queue only if the transaction was previously
```

File: src/main/java/it/pagopa/transactions/commands/handlers/TransactionSendClosureHandler.java [hunk 2]
```diff
@@ -297,26 +297,53 @@ private ClosePaymentRequestV2Dto.OutcomeEnum authorizationResultToOutcomeV2(
...
+                                            closureEvent.getTransactionId()
+                                    );
+                                    return closureEvent.getTransactionId();
+                                }
+                        )
+                )
+                .flatMap(transactionId -> {
+                    TransactionRefundRequestedEvent refundRequestedEvent = new TransactionRefundRequestedEvent(
+                            transactionId,
+                            new TransactionRefundedData(TransactionStatusDto.CLOSED)
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/commands/handlers/TransactionSendClosureHandler.java:337]: here the previous state can be both `CLOSURE_ERROR` or `CLOSED`  depending on the TaskEither being left or right, right?

Comment (giovanniberti) [reply src/main/java/it/pagopa/transactions/commands/handlers/TransactionSendClosureHandler.java:337]: You're right, fixing it

Comment (giovanniberti) [reply src/main/java/it/pagopa/transactions/commands/handlers/TransactionSendClosureHandler.java]: Fixed in a20f88f

**Review by pietro-tota — DISMISSED** [review]
🚀

**Review by pietro-tota — APPROVED** [review]

**Pull request merged**